File: objc-2/h00.override/Shape.m
#import "Shape.h"

@implementation Shape

  /******* INITIALIZERS *******/
  - (Shape *) initWithWidth: (double) w height: (double) h {
    if (self = [super init]) { width = w; height = h; }
    return self;
  }

  /******* METHODS *******/
  - (double) calculateArea { return 0.0; }

@end

File: js/ts/h00.overriding/Shape.ts
/******* RHINO/SPIDERMONKEY SUPPORT *******/
declare function print(str);
declare function load(str);
/******* CLASS DECLARATION *******/
class Shape {
  protected width :  number;
  protected height : number;

  /******* CONSTRUCTORS *******/
  constructor(width : number, height : number) {
    this.width = width;
    this.height = height;
  }

  /******* METHODS *******/
  calculateArea() : number { return 0.0; }

}
